build/Microsoft.DotNet.Cli.Test.targets

```diff
@@ -0,0 +1,157 @@
+      <TestDirectory>$(RepoRoot)/test</TestDirectory>
+      <DotnetUnderTest>$(DotnetStage2)</DotnetUnderTest>
+    </PropertyGroup>    
+
+  </Target>
+
+  <Target Name="Test" 
+          DependsOnTargets="PrepareTests;
+                            SetupTests;
+                            RestoreTests;
+                            BuildTests;
+                            RunTests;" />
+
+  <Target Name="SetupTests"
+          DependsOnTargets="SetupTestProjects;" />
+
+  <Target Name="SetupTestProjects"
+          DependsOnTargets="SetupTestPackageProjectData;
+                            RestoreTestAssetProjects;
+                            RestoreDesktopTestAssetProjects;
+                            BuildTestAssetProjects;
+                            BuildDesktopTestAssetProjects;" />
+
+  <Target Name="SetupTestPackages" 
+          DependsOnTargets="RestoreTestAssetPackages;
+                            SetupTestPackageProjectData"
+          Outputs="%(TestPackageProject.Identity)">
+    <PropertyGroup>
+      <VersionSuffixOption></VersionSuffixOption>
+      <VersionSuffixOption Condition="'%(TestPackageProject.VersionSuffix)' != ''">--version-suffix %(TestPackageProject.VersionSuffix)</VersionSuffixOption>
+    </PropertyGroup>
+    <Exec Command="$(DotnetUnderTest) pack %(TestPackageProject.FullPath) --output $(TestPackagesDir) $(VersionSuffixOption)" />
+  </Target>
+
+  <Target Name="RestoreTestAssetPackages" 
+          DependsOnTargets="PrepareTests;
+                            SetupRestoreTestAssetPackagesInputs;" 
+          Inputs="@(RestoreTestAssetPackagesInputs)"
+          Outputs="@(RestoreTestAssetPackagesInputs->'%(RelativeDir)/project.lock.json')">
+    <Exec Command="$(DotnetUnderTest) restore $(RepoRoot)/TestAssets/TestPackages/" />
+  </Target>
+
+  <Target Name="SetupRestoreTestAssetPackagesInputs">
+    <ItemGroup>
+      <RestoreTestAssetPackagesInputs Include="$(RepoRoot)/TestAssets/TestPackages/**/project.json" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="RestoreTestAssetProjects"
+          DependsOnTargets="SetupTestPackages;
+                            SetupRestoreTestAssetProjectsInputs;"
+          Inputs="@(RestoreTestAssetProjectsInputs)"
+          Outputs="@(RestoreTestAssetProjectsInputs->'%(RelativeDir)/project.lock.json')">
+    <Exec Command="$(DotnetUnderTest) restore $(RepoRoot)/TestAssets/TestProjects/ --fallbacksource $(TestPackagesDir)" />
+  </Target>
+
+  <Target Name="SetupRestoreTestAssetProjectsInputs" >
+    <ItemGroup>
+      <RestoreTestAssetProjectsInputs Include="$(RepoRoot)/TestAssets/TestProjects/**/project.json" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="CleanTestAssetProjectsLockFiles" >
+    <ItemGroup>
+      <TestAssetProjectsLockFiles Include="$(RepoRoot)/TestAssets/TestProjects/**/project.lock.json" />
+    </ItemGroup>
+    <Delete Files="@(TestAssetProjectsLockFiles)" /> 
+  </Target>
+
+  <Target Name="RestoreDesktopTestAssetProjects"
```

Comment:
I thought it would, but restore seems to work just fine. Time cost is also low, so leaving as is at the moment.